app/models/package_manager/base.rb

```diff
@@ -99,8 +99,14 @@ def self.save(project)
       end
 
       if self::HAS_VERSIONS
+        class_name = name.demodulize
         versions(project, dbproject.name).each do |version|
-          dbproject.versions.create(version) unless dbproject.versions.find { |v| v.number == version[:number] }
+          existing = dbproject.versions.find_by(number: version[:number])
```

Comment:
how bout somethin like?
```
existing = dbproject.versions.find_or_initialize_by(number: version[:number])
existing.repository_sources = (existing.repository_sources + class_name).uniq
existing.save
```
Rails won't update it if the column doesn't actually change

Reply:
Though is this actually what we want? I don't think we want to sync all versions do we?

Reply:
> Though is this actually what we want? I don't think we want to sync all versions do we?

Strike that, I see that versions(project) pulls the versions from the source data, and isn't the relationship. 👍